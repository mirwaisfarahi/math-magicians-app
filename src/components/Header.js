import React from 'react';
import { Link } from 'react-router-dom';
import {
  Nav, Navbar, NavbarBrand, NavItem, NavLink,
} from 'reactstrap';

const Header = () => (
  <Navbar className="nav">
    <NavbarBrand className="navbrand">Math magicians</NavbarBrand>
    <Nav>
      <NavItem>
        <Link to="/" className="link">
          <NavLink className="navLink"> Home </NavLink>
        </Link>
      </NavItem>
      <NavItem>
        <Link to="/calculator" className="link">
          <NavLink className="navLink"> Calculator </NavLink>
        </Link>
      </NavItem>
      <NavItem>
        <Link to="/quote" className="link">
          <NavLink className="navLink"> Quote </NavLink>
        </Link>
      </NavItem>
    </Nav>
  </Navbar>
);

export default Header;
