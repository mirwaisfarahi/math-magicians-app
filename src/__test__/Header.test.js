import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import Header from '../components/Header';

describe('test Header function', () => {
  it('Match Header DOM', () => {
    const headerTree = render(
      <Router>
        <Header />
      </Router>,
    );
    expect(headerTree).toMatchSnapshot();
  });
});
