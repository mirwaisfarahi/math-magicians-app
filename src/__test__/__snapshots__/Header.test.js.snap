// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Header function Match Header DOM 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        class="nav navbar"
      >
        <div
          class="container-fluid"
        >
          <a
            class="navbrand navbar-brand"
          >
            Math magicians
          </a>
          <ul
            class="nav"
          >
            <li
              class="nav-item"
            >
              <a
                class="link"
                href="/"
              >
                <span
                  class="navLink navbar-text"
                >
                   Home 
                </span>
              </a>
            </li>
            <li
              class="nav-item"
            >
              <a
                class="link"
                href="/calculator"
              >
                <span
                  class="navLink navbar-text"
                >
                   Calculator 
                </span>
              </a>
            </li>
            <li
              class="nav-item"
            >
              <a
                class="link"
                href="/quote"
              >
                <span
                  class="navLink navbar-text"
                >
                   Quote 
                </span>
              </a>
            </li>
          </ul>
        </div>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav
      class="nav navbar"
    >
      <div
        class="container-fluid"
      >
        <a
          class="navbrand navbar-brand"
        >
          Math magicians
        </a>
        <ul
          class="nav"
        >
          <li
            class="nav-item"
          >
            <a
              class="link"
              href="/"
            >
              <span
                class="navLink navbar-text"
              >
                 Home 
              </span>
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              class="link"
              href="/calculator"
            >
              <span
                class="navLink navbar-text"
              >
                 Calculator 
              </span>
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              class="link"
              href="/quote"
            >
              <span
                class="navLink navbar-text"
              >
                 Quote 
              </span>
            </a>
          </li>
        </ul>
      </div>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
